### PR TITLE
SNOW-1471198 move modin_sp_short_regress to modin_sp_precommit

### PR DESCRIPTION
**Pull request by sfc-gh-azhan**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1471198

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

Test only change. Note `modin_sp_short_regress` is not used any more so move to `modin_sp_precommit`
Testing passed: https://ci-dev-142.int.snowflakecomputing.com/job/SnowparkPandasStoredProcPrecommitTest/77/

### CODE REVIEW

**Review by sfc-gh-nkrishna — APPROVED** [review]

File: tests/integ/modin/frame/test_head_tail.py [hunk 1]
```diff
@@ -29,7 +29,7 @@ def eval_result_and_query_with_no_join(
     assert_snowpark_pandas_equals_to_pandas_without_dtypecheck(snow, native, **kwargs)
 
 
-@pytest.mark.modin_sp_short_regress
+@pytest.mark.modin_sp_precommit
```
Comment (sfc-gh-lmukhopadhyay) [inline tests/integ/modin/frame/test_head_tail.py:32]: @sfc-gh-azhan The `frame/test_head_tail, merge, pivot, and groupby` tests here have already been directly copied over to Snowfort short regress. Do we still need to run these in precommit or can we delete the marker completely to reduce redundency?

Comment (sfc-gh-azhan) [reply tests/integ/modin/frame/test_head_tail.py:32]: If they have been covered already, then let me just remove them.

File: tests/unit/modin/test_internal_frame.py [hunk 2]
```diff
@@ -37,7 +37,7 @@ def __init__(self, sp_df: OrderedDataFrame, internal: InternalFrame) -> None:
         self.internal_frame: InternalFrame = internal
 
 
-@pytest.mark.modin_sp_short_regress
+@pytest.mark.modin_sp_precommit
```
Comment (sfc-gh-lmukhopadhyay) [inline tests/unit/modin/test_internal_frame.py:40]: Do we still want to run these unit tests in precommit? I'm not sure how helpful they will be here

Comment (sfc-gh-azhan) [reply tests/unit/modin/test_internal_frame.py:40]: Do we have them run in other places? If not, we can run them here. They are pretty quick to run.

Comment (sfc-gh-lmukhopadhyay) [reply tests/unit/modin/test_internal_frame.py:40]: Sounds good, they're not run anywhere else so we can run them here

**Review by sfc-gh-lmukhopadhyay — APPROVED** [review]
LGTM

**Pull request merged**